splash/browser_tab.py

```diff
@@ -287,6 +289,11 @@ def go(self, url, callback, errback, baseurl=None, http_method='GET',
         """
         self.store_har_timing("_onStarted")
 
+        if body and http_method not in ["POST", "PUT"]:
+            # if someone gives body but forgets method
+            # QT will raise exceptions
+            http_method = "POST"
```

Comment:
yeah you're probably right, not needing to pass "method" everytime could sometimes be a "feature" (you can only pass body and it will default to POST) but on the other hand it can make some programmer mistakes difficult to spot - someone will pass wrong method something could go wrong but he will not know why. 

So will raise 400 with bad method here.